README.md

```diff
@@ -14,11 +14,279 @@ background](https://neilmitchell.blogspot.com/2020/01/one-haskell-ide-to-rule-th
+```
+
+**Debian 10/Ubuntu 18.10 or later**:
+
+```bash
+sudo apt install libicu-dev libncurses-dev libgmp-dev
+```
+
+**Fedora**:
+
+```bash
+sudo dnf install libicu-devel ncurses-devel # also zlib-devel if not already installed
+```
+
+#### Windows-specific pre-requirements
+
+In order to avoid problems with long paths on Windows you can do either one of the following:
+
+1. Clone the `haskell-language-server` to a short path, for example the root of your logical drive (e.g. to
+   `C:\hls`). Even if you choose `C:\haskell-language-server` you could hit the problem. If this doesn't work or you want to use a longer path, try the second option.
+
+2. If the `Local Group Policy Editor` is available on your system, go to: `Local Computer Policy -> Computer Configuration -> Administrative Templates -> System -> Filesystem` set `Enable Win32 long paths` to `Enabled`. If you don't have the policy editor you can use regedit by using the following instructions [here](https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#enable-long-paths-in-windows-10-version-1607-and-later). You also need to configure git to allow longer paths by using unicode paths. To set this for all your git repositories use `git config --system core.longpaths true` (you probably need an administrative shell for this) or for just this one repository use `git config core.longpaths true`.
+
+In addition make sure `haskell-language-server.exe` is not running by closing your editor, otherwise in case of an upgrade the executable can not be installed.
+
+#### Download the source code
+
+```bash
+git clone https://github.com/haskell/haskell-language-server --recurse-submodules
+cd haskell-language-server
+```
+
+#### Building
+
+Note, on first invocation of the build script with stack, a GHC is being installed for execution.
+The GHC used for the `install.hs` can be adjusted in `./install/shake.yaml` by using a different resolver.
+
+Available commands can be seen with:
+
+```bash
+stack ./install.hs help
+```
+
+Remember, this will take time to download a Stackage-LTS and an appropriate GHC for build
+haskell-language-server the first time.
+
+##### Install via cabal
+
+The install-script can be invoked via `cabal` instead of `stack` with the command
+
+```bash
+cabal v2-run ./install.hs --project-file install/shake.project <target>
+```
+
+or using the existing alias script
+
+```bash
+./cabal-hls-install <target>
+```
+
+Running the script with cabal on windows requires a cabal version greater or equal to `3.0.0.0`.
+
+For brevity, only the `stack`-based commands are presented in the following sections.
+
+##### Install specific GHC Version
+
+Install haskell-language-server for the latest available and supported GHC version (and hoogle docs):
+
+```bash
+stack ./install.hs hls
+```
+
+Install haskell-language-server for a specific GHC version (and hoogle docs):
+
+```bash
+stack ./install.hs hls-8.8.3
+stack ./install.hs data
```

Comment:
This target is currently broken. We should either not mention it right now or fix it in the install script

Reply:
oh, yeah, it is not needed right now, but do you think the hoohle plugin will be restored?

Reply:
I think it will. It probably makes more sense to fix the target in the install script, such that it does not attempt to populate the hoogle database

Reply:
maybe printing a warning for the specific data target?

Reply:
What did the hoogle plugin do? ghcide is already able to suggest imports for modules in the package dependencies of the project, in case it was that

Reply:
It was fallback for hover in case haddocks were not available.

Reply:
Ok, the data target has been disabled